kotlin/find-address/src/main/java/com/esri/arcgisruntime/sample/findaddress/MainActivity.kt

```diff
@@ -0,0 +1,319 @@
+      // set the map viewpoint to start over North America
+      setViewpoint(Viewpoint(40.0, -100.0, 100000000.0))
+      // define the graphics overlay and add it to the map view
+      graphicsOverlay = GraphicsOverlay()
+      graphicsOverlays.add(graphicsOverlay)
+    }
+
+    // create a locator task from an online service
+    locatorTask = LocatorTask(getString(R.string.locator_task_uri))
+
+    // create the picture marker symbol to show address location
+    createPinSymbol()
+
+    // add listener to handle screen taps
+    mapView.onTouchListener = object : DefaultMapViewOnTouchListener(this, mapView) {
+      override fun onSingleTapConfirmed(motionEvent: MotionEvent): Boolean {
+        identifyGraphic(motionEvent)
+        return true
+      }
+    }
+
+    setupAddressSearchView()
+  }
+
+  /**
+   * Sets up the address SearchView and uses MatrixCursor to show suggestions to the user as text is input.
+   */
+  private fun setupAddressSearchView() {
+    addressGeocodeParameters = GeocodeParameters().apply {
+      // get place name and street address attributes
+      resultAttributeNames.addAll(listOf("PlaceName", "StAddr"))
+      // return only the closest result
+      maxResults = 1
+
+      addressSearchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+        override fun onQueryTextSubmit(address: String): Boolean {
+          // geocode typed address
+          geoCodeTypedAddress(address)
+          // clear focus from search views
+          addressSearchView.clearFocus()
+          return true
+        }
+
+        override fun onQueryTextChange(newText: String?): Boolean {
+         // if the newText string isn't empty, get suggestions from the locator task
+          if (!newText.equals("")) {
+            val suggestionsFuture = locatorTask.suggestAsync(newText)
+            suggestionsFuture.addDoneListener{
+              try {
+                // get the results of the async operation
+                val suggestResults = suggestionsFuture.get()
+                // set up parameters for searching with MatrixCursor
+                val address = "address"
+                val columnNames = arrayOf(BaseColumns._ID, address)
+                val suggestionsCursor = MatrixCursor(columnNames)
+                var key = 0
+
+                // add each address suggestion to a new row
+                for (result in suggestResults) {
+                  suggestionsCursor.addRow(arrayOf<Any>(key++, result.getLabel()))
+                }
+                // column names for the adapter to look at when mapping data
+                val cols = arrayOf(address)
+                // ids that show where data should be assigned in the layout
+                val to = intArrayOf(R.id.suggestion_address)
+                // define SimpleCursorAdapter
+                val suggestionAdapter = SimpleCursorAdapter(this@MainActivity, R.layout.suggestion,
+                    suggestionsCursor, cols, to, 0)
```

Comment:
To me it's cleaner here since we can use the setting up of each variable to comment on what the code is doing before passing it in to `SimpleCursorAdapter`, but yes they could also be inlined. If that level of clarification isn't helpful/required, then we could inline.